core/integration/module_typescript_test.go

```diff
@@ -1329,7 +1329,330 @@ export class Test {
 `))
 
 		_, err := modGen.With(daggerQuery(`{test{bool(false)}}`)).Stdout(ctx)
-		require.ErrorContains(t, err, "Use of primitive Boolean type detected, did you mean boolean?")
+		require.ErrorContains(t, err, "Use of primitive 'Boolean' type detected, please use 'boolean' instead.")
+	})
+}
+
+func (TypescriptSuite) TestNativeEnumType(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	modGen := c.Container().From(golangImage).
+		WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+		WithWorkdir("/work").
+		With(daggerExec("init", "--name=test", "--sdk=typescript")).
+		With(sdkSource("typescript", `
+import { object, func } from "@dagger.io/dagger"
+
+/**
+ * Test Enum
+ */
+export enum TestEnum {
+    /**
+     * A
+     */
+    A = "a",
+
+    /**
+     * B
+     */
+    B = "b",
+}
+
+@object()
+export class Test {
+  @func()
+  testEnum(test: TestEnum = TestEnum.A): TestEnum {
+    return test
+  }
+}
+  `))
+
+	t.Run("native enum type - doc", func(ctx context.Context, t *testctx.T) {
+		schema := inspectModule(ctx, t, modGen)
+
+		require.Equal(t, "Test Enum", schema.Get("enums.#.asEnum|#(name=TestEnum).description").String())
+		require.Equal(t, "A", schema.Get("enums.#.asEnum|#(name=TestEnum).values.#(name=a).description").String())
+		require.Equal(t, "B", schema.Get("enums.#.asEnum|#(name=TestEnum).values.#(name=b).description").String())
+	})
+
+	t.Run("native enum type - correct input / output", func(ctx context.Context, t *testctx.T) {
+		out, err := modGen.With(daggerCall("test-enum", "--test", "b")).Stdout(ctx)
+		require.NoError(t, err)
+		require.Equal(t, `b`, out)
+	})
+
+	t.Run("native enum type - default value by reference", func(ctx context.Context, t *testctx.T) {
+		out, err := modGen.With(daggerCall("test-enum")).Stdout(ctx)
+		require.NoError(t, err)
+		require.Equal(t, `a`, out)
+	})
+}
+
+func (TypescriptSuite) TestReferencedDefaultValue(ctx context.Context, t *testctx.T) {
```

Comment:
Please add a test checking inspection for what's the actual default value (in JSON) that was registered to the API with these primitives. Same way you test for documentation, for example.

Reply:
Okay!

Reply:
Test added

Reply:
Thanks, so as you can see, if you add that cast back you get this error:
```
	Error:      	Not equal: 
	            	expected: "4"
	            	actual  : "\"4\""
	            	
	            	Diff:
	            	--- Expected
	            	+++ Actual
	            	@@ -1 +1 @@
	            	-4
	            	+"4"
	Test:       	TestTypescript/TestReferencedDefaultValue/check_default_value_in_doc
```